File: Week2/4_least_common_multiple/lcm.cpp
#include <iostream>

using namespace std;

int euclidGCD(int a, int b){

    if (b==0) return a;
    
    return euclidGCD(b, a%b);
}

long long lcm_naive(long long a, long long b){
    
    return (a*b)/euclidGCD(a, b);
}

int main(){
int a;
int b;

cout << "Enter the numbers you want to calculate the LCM: ";
cin >> a >> b;

cout << "The LCM is: " ;
cout << lcm_naive(a, b);
return 0;
}

File: Week2/4_least_common_multiple/lcm.py
def euclidGCD(a, b):
    if b==0:
        return a
    else:
        return euclidGCD(b, a%b)

def lcm_naive(a, b):

    return int((a*b) /euclidGCD(a, b))

if __name__ == '__main__':
    
    print("Enter the numbers you want to calculate LCM: ")
    a = int(input())
    b = int(input())

    print("The LCM is: ", lcm_naive(a, b))

